managedtenants/utils/ocm.py

```diff
@@ -107,16 +103,18 @@ def add_addon(self, metadata):
         return self._post("/api/clusters_mgmt/v1/addons", json=addon)
 
     def _addon_exists(self, addon_id):
-        resp = self._get(f"/api/clusters_mgmt/v1/addons/{addon_id}")
-        return resp.status_code == 200
+        try:
+            self._get(f"/api/clusters_mgmt/v1/addons/{addon_id}")
+            return True
+        # `_get` raises OCMAPIError on 404's
+        except OCMAPIError:
+            return False
 
     def add_addon_version(self, imageset, metadata):
         if self._addon_exists(metadata.get("id")):
             return self._add_addon_version(imageset, metadata)
         # Create the addon first
-        resp = self.add_addon(metadata)
-        if resp.status_code != 201:
-            raise AddonCreationError("Failed to succesfully create the addon.")
```

Comment:
We dont need a separate exception class here as `_post` method will raise `OCMAPIError` on failed requests (!= 201)